src/jit/gentree.h

```diff
@@ -2334,10 +2334,85 @@ struct GenTreeColon: public GenTreeOp
         {}
 };
 
-/* gtCall   -- method call      (GT_CALL) */
+// gtCall   -- method call      (GT_CALL)
 typedef class fgArgInfo *  fgArgInfoPtr;
 enum class InlineObservation;
 
+// Multi-reg return type descriptor of a GT_CALL node.
+// x64 Unix, Arm64, Arm32 and x86 allow a value to be returned in multiple
+// registers. For such calls this struct provides the following info 
+// on their return type
+//    - type of value returned in each return register
+//    - return register numbers in which the value is returned
+//    - a spill mask for lsra/codegen purpose
+//
+// Right now it is implemented to meet the needs of x64 unix.
+// Eventually it will be updated to meet the needs of Arm32/arm64.
+struct MultiRegReturnTypeDesc
+{
+private:
+    var_types regType0;
+    var_types regType1;
+
+#ifdef DEBUG
+    bool m_inited;
+#endif
+
+public:
+    MultiRegReturnTypeDesc()
+    {
+        regType0 = TYP_UNKNOWN;
+        regType1 = TYP_UNKNOWN;
+
+#ifdef DEBUG
+        m_inited = false;
+#endif
+    }
+
+    void Initialize(Compiler* comp, CORINFO_CLASS_HANDLE retClsHnd);
+
+    // Get the count of return registers in which the return value is returned.
+    // Returns 0 if the return type is not returned in registers.
+    int GetReturnRegCount()
+    {
+        assert(m_inited);
+
+        int regCount = 0;
+        if (regType0 != TYP_UNKNOWN)
+        {
+            ++regCount;
+
+            if (regType1 != TYP_UNKNOWN)
+            {
+                ++regCount;
+            }
+        }
+        else
+        {
+            // If regType0 is TYP_UNKNOWN then regType0 too must be TYP_UNKNOWN.
+            assert(regType1 == TYP_UNKNOWN);
+        }
+
+        return regCount;
+    }
+
+    // Get var_type of the first return register.
+    // Assumption: Caller is expected to call this method only if return reg count >=1
+    var_types GetFirstReturnRegType()
+    {
+        assert(regType0 != TYP_UNKNOWN);
+        return regType0;
+    }
+
+    // Get var_type of the second return register.
+    // Assumption: Caller is expected to call this method only if return reg count == 2
+    var_types GetSecondReturnRegType()
+    {
+        assert(regType1 != TYP_UNKNOWN);
+        return regType1;
+    }
+};
```

Comment:
Should this be more general, e.g.: GetReturningRegType(unsigned regIndex), for regIndex == 0 or 1... and possibly greater for cases of more than 2 return regs?